### DISCUSSION TITLE
Confused on why I keep getting "tag already exist"

### DISCUSSION THREAD

**Discussion by factiondavid** [semantic-release/semantic-release#3297] [Q&A]
I'm using my develop branch to do a pre-release. 

It seems as of lately when I merge to develop it keeps forgetting tags.  Any idea why? 

My current workflow. Feat > dev> pre-release > main > release. 

Shouldn't I be able to continuously merge to dev and create new pre-releases?

**Comment by travi** [comment]
you are most likely rewriting the history of the pre-release branch. the previous tag needs to exist in the git history of that branch. use normal merges

also, it is worth understanding that using the pre-release workflow to simulate gitflow is not an officially supported workflow for semantic-release. if you choose to use that approach anyway, you need to understand that we could unintentionally break you in the future

**Reply by factiondavid** [reply]
I guess I'm a bit confused as to why this approach is different than setting a beta branch? In my configuration I set dev as the prerelease alpha branch. 

I would like to understand what I'm doing that's so different than what's being discussed [here](https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration).

**Reply by travi** [reply]
pre-releases are intended to produce a version for feedback by consumers. these should hopefully be rare and short-lived for gaining feedback on specific upcoming features. once feedback is gained and maintainers are ready to promote to the stable channel, merging the branch makes that easy. once merged, the feedback pre-release branch should be removed and only re-created if/when another feature needs consumer feedback. 

pre-releases are not designed to support a workflow that requires publishing releases in order to verify that the project works as intended. we do not provide an official workflow to support such workflows. there are better options available to do behavior verification before publishing in the vein of continuous integration rather than gitflow.

all of that aside, no matter how you decide to use our release workflows, rewriting history is not supported because the intent is enable maintaining the lineage between versions.

**Reply by factiondavid** [reply]
What is the recommendend approach with dev and staging branches? 

I have a dilemma where staging and dev branches versions to be ahead of main. 

I'm using sentry.io to collection errors on my app and it gets the version from package.json. However, if main releases then it doesn't make sense for dev, and staging to stay behind main. 

Not sure what the recommendation approach is. Do I just need three release branches then?

**Reply by travi** [reply]
you can merge from main, you just need to do so without rewriting history. that means using normal merges rather than sqaush or rebase.

again, i'm wondering if semantic-release is truly the tool you are looking for. it sounds like you are deploying an application rather than a library or other dependency. is semantic versioning actually valuable to your process?
